lib/ProgramCall.js

```diff
@@ -31,59 +33,108 @@ class ProgramCall {
   }
 
   /**
-    * @description adds a parameter to the program XML
-    * @param {string | array} data
-    * @param {string} [type]
-    * @param {object} options
-    * @param {*} inDs
-    */
-  addParam(data, type, options, inDs = null) {
-    let opt;
-    // DS element has no 'type', so if the second param 'type' is an Object, then it is the options.
-    if (Array.isArray(data)) {
-      opt = type;
-    } else {
-      opt = options;
-      if (type === undefined) {
-        throw new Error('Specifying the parameter type is required.');
-      }
+   * Internal function to create <data> nodes
+   * There is an open propsal to add private methods to JS.
+   * https://github.com/tc39/proposal-private-methods
+   * We should update to use private methods in the future.
+   * @param {string} name
+   * @param {string} type
+   * @param {string} value
+   * @param {object} options
+  */
+  __addDataNode__(name, type, value, options) {
```

Comment:
I don't see why this function is needed. It should be rolled in to `__addData__`.